telemetry_api/lib/telemetry_api/traces.ex

```diff
@@ -151,6 +159,31 @@ defmodule TelemetryApi.Traces do
     end
   end
 
+  defp add_event(merkle_root, event_name, event_attributes, operator_stake) do
```

Comment:
Maybe we should refactor this. Since the part we are repeating everywhere is the trace retrieval and setting it as current, we could introduce a method called `set_current_trace(merkle_root)` to handle that. Then, we could directly use the `Tracer.add_event()`. WDYT?

Reply:
We will do it in another PR